src/shared/ui/SearchForm/SearchForm.test.js

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import "@testing-library/jest-dom";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { SearchForm } from "./SearchForm";
+import { keyboard } from "@testing-library/user-event/dist/keyboard";
+
+describe("SearchForm Component Functionality", () => {
+  test("Should Render Initial Value", () => {
+    const initialValue = "Test";
+    render(<SearchForm initialValue={initialValue} />);
+    const inputElement = screen.getByRole("textbox");
+    const inputValue = inputElement.value;
+    expect(inputValue).toBe(initialValue);
+  });
+  test("Should onChange fire with right value when clicked submit after typing", () => {
+    const mockOnChange = jest.fn((e) => e.preventDefault());
+    const changedValue = "Javascript";
+
+    render(<SearchForm onChange={mockOnChange} />);
+
+    const input = screen.getByRole("textbox");
+    const button = screen.getByRole("button");
+
+    userEvent.type(input, changedValue);
+    userEvent.click(button);
+
+    // This part doesn't work, I didn't find a way to pass an event argument
+
+    // expect(mockOnChange).toHaveBeenCalledWith(
```

Comment:
userEvent.type(input, changedValue);
```suggestion
    // expect(mockOnChange).toHaveBeenCalledWith(
```

Reply:
I have changed based on the first comment